### PR TITLE
Matcher for "result of comparison against a string literal is unspecified"

### PR DESCRIPTION
**Pull request by brianyu28**
Addressing #62, which appears in `/review` inputs `227` and `2139`.

### CODE REVIEW

File: helpers/clang.py [hunk 1]
```diff
@@ -366,6 +366,22 @@ def help(lines):
...
     # /tmp/foo-1ce1b9.o: In function `main':
+    # foo.c:6:14: error: result of comparison against a string literal is unspecified (use strncmp instead) [-Werror,-Wstring-compare]
+    #     if (word < "twenty-eight")
+    #              ^ ~~~~~~~~~~~~~~
+    matches = re.search(r"^([^:]+):(\d+):\d+: (?:warning|error): result of comparison against a string literal is unspecified", lines[0])
+    if matches:
+        response = [
+            "You can't compare two strings the same way you would compare two numbers (with `<`, `>`, etc.).",
+            "Did you mean to compare two characters instead? If so, try using single quotation marks around characters instead of double quotation marks.",
+            "If you need to compare two strings, try using the `strcmp` function declared in `<string.h>`."
```
Comment (dmalan) [inline helpers/clang.py]: No need for brackets around header files' names.

File: helpers/clang.py [hunk 2]
```diff
@@ -366,6 +366,22 @@ def help(lines):
     
     # $ clang foo.c
     # /tmp/foo-1ce1b9.o: In function `main':
+    # foo.c:6:14: error: result of comparison against a string literal is unspecified (use strncmp instead) [-Werror,-Wstring-compare]
+    #     if (word < "twenty-eight")
+    #              ^ ~~~~~~~~~~~~~~
+    matches = re.search(r"^([^:]+):(\d+):\d+: (?:warning|error): result of comparison against a string literal is unspecified", lines[0])
+    if matches:
+        response = [
+            "You can't compare two strings the same way you would compare two numbers (with `<`, `>`, etc.).",
```
Comment (dmalan) [inline helpers/clang.py]: Lacks mention of the filename / line number in question?

**Comment by brianyu28** [comment]
Fixed.

**Comment by dmalan** [comment]
:shipit:

**Pull request merged**